Source/Core/CameraLimiter.js

```diff
@@ -0,0 +1,132 @@
+    '../Core/HeadingPitchRoll',
+    '../Core/Matrix4',
+    '../Core/Quaternion',
+    '../Core/Transforms'
+], function(
+    Cartesian3,
+    Check,
+    defaultValue,
+    defined,
+    HeadingPitchRoll,
+    Matrix4,
+    Quaternion,
+    Transforms) {
+    'use strict';
+
+    /**
+     * The CameraLimiter is used to limit a {@link Camera}'s position (using a bounding object) and orientation (defined by min and max values of {@link HeadingPitchRoll}).
+     *
+     * If set, the camera of this limiter will be constrained by location and viewing ability for moving and changes in the look vector.
+     * The camera ignores position and orientation limits in {@link SceneMode#SCENE2D} view and {@link SceneMode#COLUMBUS_VIEW}.
+     *
+     * @alias CameraLimiter
+     * @constructor
+     *
+     * @param {AxisAlignedBoundingBox|BoundingSphere|OrientedBoundingBox} [options.boundingObject] Assigned to the boundingObject attribute which can be used to specify limits on locations.
+     * @param {HeadingPitchRoll} [options.minimumHeadingPitchRoll] Assigned to the minHeadingPitchRoll attribute which can be used to limit the minimum orientation.
+     * @param {HeadingPitchRoll} [options.maximumHeadingPitchRoll] Assigned to the maxHeadingPitchRoll attribute which can be used to limit the maximum orientation.
+     */
+    function CameraLimiter(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        /**
+         * The BoundingObject. Used to restrict camera positions of the {@link Camera} holding this CameraLimiter.
+         * @type {AxisAlignedBoundingBox|BoundingSphere|OrientedBoundingBox}
+         * @default undefined
+         */
+        this.boundingObject = options.boundingObject;
+
+        /**
+         * The minimum orientation values.
+         * @type {HeadingPitchRoll}
+         */
+        this.minHeadingPitchRoll = HeadingPitchRoll.clone(options.minimumHeadingPitchRoll);
+
+        /**
+         * The maximum orientation values.
+         * @type {HeadingPitchRoll}
+         */
+        this.maxHeadingPitchRoll = HeadingPitchRoll.clone(options.maximumHeadingPitchRoll);
+    }
+
+    /**
+     * @private
+     */
+    CameraLimiter.prototype.limitPosition = function(position, result) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.object('position', position);
+        Check.typeOf.object('result', result);
+        //>>includeEnd('debug');
+
+        if (defined(this.boundingObject)) {
+            result = this.boundingObject.projectedPoint(position, result);
+        }
+        return result;
+    };
+
+    var scratchMatrix4 = new Matrix4();
+    /**
+     * @private
+     */
+    CameraLimiter.prototype.limitOrientation = function(orientation, position, result) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.object('orientation', orientation);
+        Check.typeOf.object('result', result);
+        //>>includeEnd('debug');
```

Comment:
You don't need to add these checks for private functions, same as above with `limitPosition`